pkg/support/dump_test.go

```diff
@@ -15,21 +15,83 @@
 package support
 
 import (
+	"fmt"
+	"path/filepath"
+	"strings"
 	"testing"
+	"time"
 
+	"github.com/spf13/afero"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/utils/exec"
+	exectesting "k8s.io/utils/exec/testing"
+
+	"antrea.io/antrea/pkg/util/runtime"
 )
 
-func TestParseLogDate(t *testing.T) {
-	data := "I0817 06:55:10.804384       1 shared_informer.go:270] caches populated"
-	ts, err := parseTimeFromLogLine(data, "2021", "antrea-agent")
-	assert.Nil(t, err)
-	assert.Equal(t, ts.String(), "2021-08-17 06:55:10 +0000 UTC")
+type testExecuter struct {
+	exectesting.FakeExec
+}
 
-	data = "2021-06-01T09:30:43.823Z|00004|memory|INFO|cells:299 monitors:2 sessions:2"
-	ts, err = parseTimeFromLogLine(data, "2021", "ovs")
-	assert.Nil(t, err)
-	assert.Equal(t, ts.String(), "2021-06-01 09:30:43 +0000 UTC")
+func (te *testExecuter) Command(cmd string, args ...string) exec.Cmd {
+	fakeCmd := new(exectesting.FakeCmd)
+	fakeCmd.CombinedOutputScript = append(fakeCmd.CombinedOutputScript, func() ([]byte, []byte, error) {
+		return []byte(fmt.Sprintf("%s %s", cmd, strings.Join(args, " "))), nil, nil
+	})
+	return fakeCmd
+}
+
+func TestControllerDumpLog(t *testing.T) {
+	basedir := "dir1/dir2"
+	fs := afero.NewOsFs()
+	varr := NewControllerDumper(fs, nil, "7s")
+	err := varr.DumpLog(basedir)
+	assert.NoError(t, err)
+}
+
+func TestParseTimeFromLogLine(t *testing.T) {
+	tests := []struct {
+		name              string
+		data              string
+		year              string
+		prefix            string
+		expectedError     string
+		expectedTimeStamp string
+	}{
+		{
+			name:              "log line with prefix antrea-agent",
+			data:              "I0817 06:55:10.804384       1 shared_informer.go:270] caches populated",
+			year:              "2021",
+			prefix:            "antrea-agent",
+			expectedTimeStamp: "2021-08-17 06:55:10 +0000 UTC",
+		},
+		{
+			name:              "log line with prefix ovs",
+			data:              "2021-06-01T09:30:43.823Z|00004|memory|INFO|cells:299 monitors:2 sessions:2",
+			year:              "2021",
+			prefix:            "ovs",
+			expectedTimeStamp: "2021-06-01 09:30:43 +0000 UTC",
+		},
+		{
+			name:          "with no log line",
+			year:          "2021",
+			prefix:        "ovs",
+			expectedError: "log line is empty",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			gotTimeStamp, err := parseTimeFromLogLine(tc.data, tc.year, tc.prefix)
+			if tc.expectedError != "" {
+				require.EqualError(t, err, tc.expectedError)
+			} else {
+				assert.Equal(t, gotTimeStamp.String(), tc.expectedTimeStamp)
+				assert.NoError(t, err)
```

Comment:
you should always check for no error first, usually it is with `require.NoError` and not `assert.NoError`.